src/pspm_find_valid_fixations.m

```diff
@@ -153,312 +148,248 @@
     end
-  end
-end
-op = struct();
-op.overwrite = options.overwrite;
-if ~isempty(options.newfile)
-  [pathstr, ~, ~] = fileparts(options.newfile);
-  if exist(pathstr, 'dir') || isempty(pathstr)
-    out_file = options.newfile;
-  else
-    warning('ID:invalid_input', 'Path to options.newfile (%s) does not exist.', options.newfile);
-  end
-else
-  out_file = fn;
-end
-% collect data
-if options.missing
-  new_chans = [[new_excl{:}], [new_pu{:}]];
-else
-  new_chans = [new_pu{:}];
-end
-if numel(new_chans) >= 1
-  new_data = data;
-  chan_idx = NaN(1,numel(new_chans));
-  for i = 1:numel(new_chans)
-    if strcmpi(options.channel_action, 'add')
-      new_data{end+1} = new_chans{i};
-      chan_idx(i) = numel(new_data);
+    if (rsts(1) < 1 && rsts(2) < 1) 
+        return; 
+    elseif (rsts(1) < 1 || rsts(2) < 1)
+        pos_of_channel(rsts < 1) = [];
     else
-      % look for same chan_type
-      channel = cellfun(@(x) strcmpi(new_chans{i}.header.chantype, x.header.chantype), new_data);
-      if any(channel)
-        % replace the first found channel
-        idx = find(channel, 1, 'first');
-        new_data{idx}.data = new_chans{i}.data;
-        chan_idx(i) = idx;
-      else
-        new_data{end+1} = new_chans{i};
-        chan_idx(i) = numel(new_data);
-      end
+        % update best eye
+        eye_stat = Inf(1,numel(alldata.infos.source.eyesObserved));
+        for i = 1:numel(alldata.infos.source.eyesObserved)
+            e_stat = alldata.infos.source.chan_stats(pos_of_channel);
+            eye_stat(i) = max(cellfun(@(x) x.nan_ratio, e_stat));
+        end
+        [~, min_idx] = min(eye_stat);
+        alldata.infos.source.best_eye = lower(alldata.infos.source.eyesObserved(min_idx));
     end
-  end
-  % update channel stats (similar to pspm_get_eyelink)
-  for i = 1:numel(new_data)
-    % update nan ratio
-    n_inv = sum(isnan(new_data{i}.data));
-    n_data = numel(new_data{i}.data);
-    infos.source.chan_stats{i}.nan_ratio = n_inv/n_data;
-  end
-  % update best eye
-  eye_stat = Inf(1,numel(infos.source.eyesObserved));
-  for i = 1:numel(infos.source.eyesObserved)
-    e = lower(infos.source.eyesObserved(i));
-    e_stat = {infos.source.chan_stats{...
-      cellfun(@(x) ~isempty(regexpi(x.header.chantype, ['_' e], 'once')), new_data)}};
-    eye_stat(i) = max(cellfun(@(x) x.nan_ratio, e_stat));
-  end
-  [~, min_idx] = min(eye_stat);
-  infos.source.best_eye = lower(infos.source.eyesObserved(min_idx));
-  file_struct.infos = infos;
-  file_struct.data = new_data;
-  file_struct.options = op;
-  [sts_load_data, ~, ~, ~] = pspm_load_data(out_file, file_struct);
-else
-  warning('ID:invalid_input', 'Appearently no data was generated.');
 end
-sts = 1;
-return
+
+% write to file or return data
+if ischar(fn)
+  alldata.options = struct('overwrite', 1); 
+  [sts, ~, ~, ~] = pspm_load_data(fn, alldata);
+elseif isstruct(fn)
+    sts = 1;
+    fn = alldata;
```

Comment:
I dont quite understand here... `alldata` are data channels, and `fn` is the path to the file. I can understand if `fn` is a struct, like a list of several file paths. I dont quite understand why `alldata` can be directly transferred to `fn`?

Reply:
With this syntax, `pspm_load_data` saves the data contained in the struct `alldata` to the file `fn`.